lms/djangoapps/instructor_task/tests/test_integration.py

```diff
@@ -134,8 +149,35 @@ def submit_rescore_one_student_answer(self, instructor, problem_url_name, studen
                                                   InstructorTaskModuleTestCase.problem_location(problem_url_name),
                                                   student)
 
-    def test_rescoring_option_problem(self):
-        """Run rescore scenario on option problem"""
+    RescoringTestCase = namedtuple('RescoringTestCase', 'edit, new_expected_scores, new_expected_max')
+
+    @ddt.data(
+        RescoringTestCase(
+            edit=dict(correct_answer=OPTION_2), new_expected_scores=(0, 1, 1, 2), new_expected_max=2,
+        ),
+        RescoringTestCase(
+            edit=dict(num_inputs=2), new_expected_scores=(2, 1, 1, 0), new_expected_max=4,
+        ),
+        RescoringTestCase(
+            edit=dict(num_inputs=4), new_expected_scores=(2, 1, 1, 0), new_expected_max=8,
+        ),
+        RescoringTestCase(
+            edit=dict(num_responses=4), new_expected_scores=(2, 1, 1, 0), new_expected_max=4,
+        ),
+        RescoringTestCase(
+            edit=dict(num_inputs=2, num_responses=4), new_expected_scores=(2, 1, 1, 0), new_expected_max=8,
+        ),
```

Comment:
I like it too, but I'd recommend naming the tuple something other than "TestCase", since that's terminology that `unittest` uses for naming classes that hold tests.